mautic/mautic
# Catch ClientException's to account for validation errors.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Y
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
We were not catching 4xx client errors when processing form repost actions. This resulted in errors from the reposted form not bubbling up to the original form.


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a form with a repost action that points to an endpoint which returns a 4xx error when validation fails. You can use https://mega-beta.mautic.com/api/user/register as that endpoint for the form.
3. When creating the form, have email & subdomain fields only (to make it easy)
4. Access the form by navigating to https://your-mautic.local/form/FORMID where FORMID is the ID of the form you just created.
5. Fill out and submit the form, leaving the subdomain blank
6. See that you get a subdomain is required error under that field
